test/cloudtest/test_k8s_node_recovery.py

```diff
@@ -0,0 +1,387 @@
+
+    # re-run testdrive to make sure it connects to the most recent envd
+    for run in range(0, int(timeout_in_sec / testdrive_timeout_in_sec)):
+        try:
+            mz.testdrive.run(
+                input=dedent(
+                    f"""
+                    > SELECT COUNT(*) {comparison_operator} {reached_index} FROM source; -- validate source
+                    true
+    
+                    > SELECT COUNT(*) {comparison_operator} {reached_index} FROM mv; -- validate mv
+                    true
+                    """
+                ),
+                default_timeout=f"{testdrive_timeout_in_sec}s",
+                no_reset=True,
+            )
+            validation_succeeded = True
+            break
+        except Exception as e:
+            # arbitrary error can occur if envd is not yet ready after restart
+            print("Error occurred, retrying.")
+            last_error_message = str(e)
+
+    end_time = time.time()
+
+    if not validation_succeeded:
+        raise RuntimeError(
+            f"Failed to achieve '{expected_state}' within {timeout_in_sec}s! {last_error_message}"
+        )
+
+    duration = end_time - start_time
+    print(
+        f"Succeeded to achieve '{expected_state}' within {duration} seconds (limit: {timeout_in_sec}s)"
+    )
+
+
+def get_current_counter_index(mz: MaterializeApplication) -> int:
+    """
+    This query has no timeout. Only use it if is expected to deliver.
+    """
+    reached_value: int = mz.environmentd.sql_query("SELECT COUNT(*) FROM source")[0][0]
+    return reached_value
+
+
+def suspend_node_of_replica(
+    mz: MaterializeApplication, cluster: ClusterDefinition, replica: ReplicaDefinition
+) -> str:
+    # TODO cloud#6830: pick node name of specified replica
+    node_names = mz.get_cluster_node_names(cluster.name)
+    assert len(node_names) > 0
+    print(f"Cluster {cluster.name} uses nodes {node_names}")
+
+    suspended_node_name = node_names[0]
+    mz.suspend_k8s_node(suspended_node_name)
+    return suspended_node_name
+
+
+# @pytest.mark.long
+# @unittest.skip("Requires #20299 to work")
+def test_unreplicated_storage_cluster_on_failing_node(
+    mz: MaterializeApplication,
+) -> None:
+    """
+    An unreplicated storage cluster is on the failed node. Queries of a downstream index in serializable mode should
+    continue to work but return stale data. Staleness should resolve within a minute or two.
+    """
+    compute_cluster = default_compute_cluster()
+    storage_cluster = default_storage_cluster()
+
+    populate(mz, compute_cluster, storage_cluster)
+    reached_index = get_current_counter_index(mz)
+
+    suspended_node_name = suspend_node_of_replica(
+        mz, storage_cluster, storage_cluster.replica_definitions[0]
+    )
+
+    validate_state(
+        mz,
+        reached_index,
+        must_exceed_reached_index=False,
+        timeout_in_sec=30,
+        expected_state="stale data being delivered timely",
```

Comment:
For the `validate_state` to be checking whether stale data is being delivered timely, it needs to run in `SERIALIZABLE` mode, and not `STRICT SERIALIZABLE`, the default. Under the default `STRICT SERIALIZABLE`, the `SELECT` will hang until recovery is cimplete. So the output will read:

```
Succeeded to achieve 'stale data being delivered timely' within 82.26484894752502 seconds (limit: 300s)
```

In other words, since it is not testing the return of stale data, the response is not timely at all.

Reply:
The reason above is why the timeout =30 is too small. So I suggest doing two separate validations:
- with SERIALIZABLE, timeout_in_secs=10 , must_exceed = False
- with STRICT SERIALIZABLE, timeout_in_secs=300, must_exceed=True